.circleci/config.yml

```diff
@@ -80,9 +80,29 @@ jobs:
             tar -xvzf "go-ipfs_${IPFS_VERSION}_${IPFS_BUILD}.${IPFS_EXT:-tar.gz}"
             ${EXEC:-sudo bash} ./go-ipfs/install.sh
 
+      - run:
+          name: Init tools
+          command: |
+            make init
+
+      - run:
+          name: Install Pre-commit
+          command: |
+            make install-pre-commit
+
+      - run:
+          name: Cache Precommit
+          command: |
+            cp .pre-commit-config.yaml pre-commit-cache-key.txt
+            poetry run python --version --version >> pre-commit-cache-key.txt
+
+      - restore_cache:
+          keys:
+            - v1-pc-cache-{{ checksum "pre-commit-cache-key.txt" }}
+
```

Comment:
Can we run pre-commit in the `lint` CI job? Because we presumably don't need to run it on every GOOS_GOARCH combo, and the `lint` job is currently quite short so doing it there won't increase overall build times.